src/convertToObject.js

```diff
@@ -12,7 +12,33 @@
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  let sourceArray = [];
+
+  sourceArray = sourceString.split(';');
+
+  let trimmedSourceArray = [...sourceArray];
```

Comment:
You created this array. Don't need to make copy before each operation.
Moreover, methods like `filter` or `map` don't mutate original array.